api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -56,3 +56,83 @@ enum FaultInjectionCluster {
   // Faults are injected to ingesss traffic by an upstream service(s).
   FAULTINJECTIONCLUSTER_UPSTREAM = 2;
 }
+
+message Config {
+  oneof fault {
+    option (validate.required) = true;
+
+    Abort abort = 1;
+    Latency latency = 2;
+  }
+}
+
+message Abort {
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  AbortStatus abort_status = 3 [ (validate.rules).message.required = true ];
+}
+
+message Latency {
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  LatencyDuration latency_duration = 3 [ (validate.rules).message.required = true ];
+}
+
+message FaultPercentage {
+  float percentage = 1 [ (validate.rules).float = {gt : 0.0, lte : 100.0} ];
+}
+
+message AbortStatus {
+  int32 http_status_code = 1 [ (validate.rules).int32 = {gt : 99, lt : 600} ];
```

Comment:
we can mate it uint32 but I decided to go with int32 and validation rules - what do you think @snowp ?

Reply:
I would opt for unsigned unless there is a reason to keep it signed